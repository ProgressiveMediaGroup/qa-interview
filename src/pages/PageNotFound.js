// Node_Modules
import React, { Component } from 'react'

// Display Components
import Body from '../components/Body'

class PageNotFound extends Component {
  render () {
    return (
      <Body heading='404' subheading='Page not found'>
        <p>Sorry, but the page you were looking for has not been found.</p>
      </Body>
    )
  }
}

export default PageNotFound
