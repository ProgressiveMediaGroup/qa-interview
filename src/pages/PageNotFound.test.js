/* global it, expect, describe */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import PageNotFound from './PageNotFound'

it('renders without crashing', () => {
  shallow(
    <PageNotFound />
  )
})

it('should match snapshot', () => {
  const wrapper = shallow(
    <PageNotFound />
  )
  expect(wrapper).toMatchSnapshot()
})

describe('Testing Strapline', function () {
  it('H1 Title is displayed', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    )
    expect(wrapper.find('h1').length).toBe(1)
  })

  it('H1 Title is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    )
    expect(wrapper.find('h1').text()).toBe('404')
  })

  it('H2 Title is displayed', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    )
    expect(wrapper.find('h2').length).toBe(1)
  })

  it('H2 Title is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    )
    expect(wrapper.find('h2').text()).toBe('Page not found')
  })
})

describe('Testing Content', function () {
  it('P tag is displayed', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    )
    expect(wrapper.find('p').length).toBe(1)
  })

  it('P tag is correct', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    )
    expect(wrapper.find('p').text()).toBe('Sorry, but the page you were looking for has not been found.')
  })
})
