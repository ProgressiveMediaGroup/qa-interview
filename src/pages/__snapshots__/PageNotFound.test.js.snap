// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <p>
        Sorry, but the page you were looking for has not been found.
</p>,
      "heading": "404",
      "subheading": "Page not found",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sorry, but the page you were looking for has not been found.",
      },
      "ref": null,
      "rendered": "Sorry, but the page you were looking for has not been found.",
      "type": "p",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <p>
          Sorry, but the page you were looking for has not been found.
</p>,
        "heading": "404",
        "subheading": "Page not found",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, but the page you were looking for has not been found.",
        },
        "ref": null,
        "rendered": "Sorry, but the page you were looking for has not been found.",
        "type": "p",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
